conans/client/rest/uploader_downloader.py

```diff
@@ -15,15 +15,17 @@ def __init__(self, requester, output, verify, chunk_size=1000):
         self.requester = requester
         self.verify = verify
 
-    def upload(self, url, abs_path, auth=None, dedup=False, retry=1, retry_wait=0):
+    def upload(self, url, abs_path, auth=None, dedup=False, retry=1, retry_wait=0, headers=None):
         if dedup:
-            headers = {"X-Checksum-Deploy": "true",
-                       "X-Checksum-Sha1": sha1sum(abs_path)}
-            response = self.requester.put(url, data="", verify=self.verify, headers=headers,
+            dedup_headers = {"X-Checksum-Deploy": "true", "X-Checksum-Sha1": sha1sum(abs_path)}
+            if headers:
+                dedup_headers.update(headers)
+            response = self.requester.put(url, data="", verify=self.verify, headers=dedup_headers,
```

Comment:
What is the meaning of sending headers for dedup? Will the metadata be overwritten/updated?